src/plugins/interactive_setup/public/manual_configuration_form.tsx

```diff
@@ -0,0 +1,436 @@
+            }
+          );
+        }
+      }
+
+      if (!errors.host) {
+        try {
+          const data = await pingCluster(values);
+          if (data instanceof Error) {
+            throw data;
+          }
+          const { statusCode, certificateChain } = data;
+          requiresAuth = statusCode === 401;
+          needsToUploadCert = !certificateChain?.length;
+        } catch (error) {
+          errors.host = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.hostConnectionError',
+            {
+              defaultMessage: "Can't connect to this cluster.",
+            }
+          );
+        }
+      }
+
+      if (requiresAuth) {
+        if (!values.username) {
+          errors.username = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.usernameRequiredError',
+            {
+              defaultMessage: 'Enter a username.',
+            }
+          );
+        } else if (values.username === 'elastic') {
+          errors.username = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.usernameReservedError',
+            {
+              defaultMessage: "User 'elastic' can't be used as Kibana system user.",
+            }
+          );
+        }
+
+        if (!values.password) {
+          errors.password = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.passwordRequiredError',
+            {
+              defaultMessage: `Enter a password.`,
+            }
+          );
+        }
+      }
+
+      if (requiresCert && !values.caCert) {
+        if (needsToUploadCert) {
+          errors.caCert = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.caCertRequiredError',
+            {
+              defaultMessage: 'Upload a trusted CA certificate.',
+            }
+          );
+        } else {
+          errors.caCert = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.caCertConfirmationRequiredError',
+            {
+              defaultMessage: 'Confirm that you recognize and trust this certificate.',
+            }
+          );
+        }
+      }
+
+      return errors;
+    },
+    onSubmit: async (values) => {
+      try {
+        await http.post('/internal/interactive_setup/configure', {
+          body: JSON.stringify({
+            hosts: [values.host],
+            username: values.username,
+            password: values.password,
+            caCert: values.caCert,
+          }),
+        });
+        onSuccess?.();
+      } catch (error) {
+        if (error.body?.message.includes('self signed certificate in certificate chain')) {
+          form.setError('caCert', 'Certificate is not valid for this cluster.');
+        } else if (error.body?.message.includes('unable to authenticate user')) {
+          form.setError('password', 'Invalid username or password.');
+        } else {
+          throw error;
+        }
```

Comment:
__question:__ wondering if it'd make sense to analyze raw ES errors on the server side and return easily-recognizable and predefined error types/codes to our UI?